osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -426,7 +426,8 @@ public void FlushPendingFilterOperations()
             if (PendingFilter?.Completed == false)
             {
                 applyActiveCriteria(false);
-                Update();
+                if (IsLoaded) // Update requires Clock to be initialized
+                    Update();
```

Comment:
Actually, this code is 3 years old (from https://github.com/ppy/osu/pull/2289)

There's a test (``TestScenePlaySongSelect.TestChangeBeatmapBeforeEnter``) that's testing that exact behaviour and it doesn't fail when I remove this line... so I think it's safe to remove it.

Reply:
it might be 3 years old, but it has never depended on an actual readout of the clock being correct before.

Reply:
Yeah, but I'm thinking that removing this should be fine.

Reply:
possibly? not sure. enough stuff has changed that it's plausible that guard is not needed anymore. tried a few times to use the terse description of the repro scenario in [this comment](https://github.com/ppy/osu/pull/2289#issuecomment-375599378) but saw no discernible breakage.